examples/kafka_message_consumer_group_service/kafka_message_consumer_group_service.bal

```diff
@@ -25,13 +25,17 @@ service kafkaService on consumer {
 }
 
 function processKafkaRecord(kafka:ConsumerRecord kafkaRecord) {
-    byte[] serializedMsg = kafkaRecord.value;
-    string|error msg = strings:fromBytes(serializedMsg);
-    if (msg is string) {
-        // Print the retrieved Kafka record.
-        io:println("Topic: ", kafkaRecord.topic, " Partition: ",
-            kafkaRecord.partition.toString(), " Received Message: ", msg);
+    anydata serializedMsg = kafkaRecord.value;
+    if (serializedMsg is byte[]) {
+        string|error msg = strings:fromBytes(serializedMsg);
+        if (msg is string) {
+            // Print the retrieved Kafka record.
+            io:println("Topic: ", kafkaRecord.topic, " Partition: ",
+                kafkaRecord.partition.toString(), " Received Message: ", msg);
+        } else {
+            log:printError("Error occurred while converting message data", msg);
```

Comment:
We don't generally use a mix of logs and println. See 4 in https://github.com/ballerina-platform/ballerina-lang/blob/master/docs/bbe-guidelines/bbe-guidelines.md#general-best-practices

Reply:
But maybe we can open an issue and fix it later, as it's not a major issue

Reply:
Created an issue: https://github.com/ballerina-platform/ballerina-lang/issues/21850